custom_components/tesla_custom/switch.py

```diff
@@ -49,6 +54,32 @@ def is_on(self):
         """Get whether the switch is in on state."""
         return self.tesla_device.get_steering_wheel_heat()
 
+    async def wait_for_climate(self):
+        """Wait for Aircon."""
+
+        entry_data = self.hass.data[TESLA_DOMAIN][self.config_entry_id]
+        climate_devices = entry_data["devices"].get("climate", [])
+
+        if len(climate_devices) > 0:
+            climate_device = climate_devices[0]
+        else:
+            return False
+
+        async with async_timeout.timeout(30):
+            while True:
+                hvac_mode = climate_device.is_hvac_enabled()
+
+                if hvac_mode is True:
+                    _LOGGER.debug("HVAC Enabled")
+                    return True
+                else:
+                    _LOGGER.info("Enabing Climate to activate Heated Steering Wheel")
```

Comment:
Does this mean it will repeatedly send the climate on command every second for up to 30s?
Wouldn't you rather send it once and wait? Or is it sometimes flaky, and the 30 retries be potentially useful?

Reply:
Yes, this is probably not the best way to implement because it will spam the api. Thinking about this more, it's probably much cleaner to have a single api command that sends the hvac start command, checks ever few seconds to see if it's been enabled, and then issues the second command.

Reply:
This really feels like the wakeup decorator. Instead of waking up the car, it instead is enabling the hvac.